### ISSUE TITLE
respin_2.0.6-0-73-ubuntu16.04.1_amd64.deb iso boots to busybox in VM - UBIQUITYDESKTOP cause?

### ISSUE THREAD

**Issue by Lew-Rockwell-Fan** [ch1x0r/LinuxRespin#27]
Editing to clarify, mea culpa:

Summary:

In a baremetal system,  booted from an internal hard drive with a minimalist Ubuntu 16.04, 64 bit, made from the mini-iso with X and Openbox, no display manager, & no DE beyond Openbox, I used:
 `sudo respin clean`
 followed by
`sudo respin backup`
to make an iso.

Booting the iso in a VM under VirtualBox (Oracle's version, not Ubuntu's) failed. 

I've repeated this cycle, trying slght variations in procedure, several times. The general outline above, and the error messages I get on booting in the VM, have remained constant.
Details:
Trying to boot the iso made in a Virtualbox VM, I get the normal Respin menu, and press enter.
Then I get:

`[     2.000001] Failed to access perfecter msr (MSR c0010007 is 0)`
I think I see that or a similar message at the same time on older live disks made the same way to DO boot.

Then the Ubuntu 16.04 splash screen (Plymouth?) starts. I press escape and see this:

`mount: mounting /dev/sr0 on /cdrom failed: No such device`
[repeated many times]

FWIW, on the bare-metal system the iso was made from, dvds ARE represented as /dev/sr0

Then I get BusyBox:

```
BusyBox v1.22.1 . . . [etc., etc.]
(initramfs)"
```

There is also a line that I swear sometimes appears BEFORE  "BusyBox . . ." and sometimes AFTER (initramfs):

`Unable to find a medium containing a live filesystem.`

This is on a 64 bit 16.04 system built from the mini.iso with X & Openbox. No DM, I use startx. No DE beyond Openbox itself. Like a Lubuntu that went on Atkins and amphetamines. Pinguy and the other respin at http://ppa.launchpad.net/sergiomejia666/respin/ubuntu/
failed in pretty much the same way. It's been a couple of months maybe since I last made a live disk backup like this and at that time my kernel was, at the highest, 4.4.0-75, and might have been lower. Now it is 4.4.0-78. The iso I get from Respin is on the order of 3 giB, so the size shouldn't be a problem.

So, is it worth trying to make a DVD anyway and see if this is just a problem in the VM or is that a waste of time and plastic doomed to failure? Any other suggestions on how I might make this work? Any kind of test I can do to help fix this?

`- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -`

I'm not sure if any of these are relevant, but I've identified and tried to fix the following POTENTIAL problems:

`- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -`

The variable UBIQUITYDESKTOP is not being set on line 504. In other words, it's value, tested by inserting
`read -p "DEBUGGING: UBIQUITYDESKTOP is $UBIQUITYDESKTOP"`
after that, is "".
This leads to an error on line 507
`sed -i "s/RELEASE/$LIVECDLABEL/" "/$LIVEHOME/Desktop/`basename $UBIQUITYDESKTOP`" &> /dev/null`
which brings forth
`basename: missing operand`

I think this is because this line:

`UBIQUITYDESKTOP=`find /usr -name ubiquity*.desktop``

should be:

`UBIQUITYDESKTOP=`find /usr/share/applications -name ubiquity*.desktop``

but I'd appreciate confirmation or refutation on that. For now I've changed it and those error messages disappeared, but the booting problem is unchanged.

`- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -`

There is no

/cdrom

directory for /dev/sr0 to mount on, in either my bare metal model or in /home/respin/respin/dummysys/

There IS a directory:

/home/respin/respin/dummysys/media/cdrom

I can't find any line that looks like

mount /dev/sr0 /cdrom

or I would try changing it. Ditto for /home/respin/respin/dummysys/etc/fstab.

Now I did 
sudo mkdir  /cdrom/ 
sudo chmod ugo+rwx /cdrom

Those perms are probably too liberal but it ought to do for testing.

Didn't make any difference. Booting the iso still fails exactly the same way.

`- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -`
Because it is the same error message, based on the answer here:

https://askubuntu.com/questions/770016/cant-find-my-dvd-drive-in-ubuntu-16-04

I also tried this on the baremetal system before running respin:

    sudo add-apt-repository ppa:brandonsnider/cdrtools
    sudo apt update
    there were 3 packages so i did: sudo apt-get dist-upgrade
    sudo apt install cdrecord mkisofs
    reboot
    
Didn't make any difference. I haven't reverted this yet.

`- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -`
/etc/casper.conf
had a line
export HOST="" with my user name, NOT the hostname between the quote marks.
I made a copy of casper.conf with the same permissions and changed it to the value set for HOST to the hostname of the baremetal system. Then i commented out:
`cp /etc/casper.conf $WORKDIR/dummysys/etc/`
and replaced it with
`cp /etc/casper.conf-Alt-model $WORKDIR/dummysys/etc/casper.conf`
Still fails to boot with the same messages. Now I reversed the commenting, making the original line active again. Might try it again later if I get this to work.

`- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -`

The xresprobe here,
0.4.24ubuntu11-0-1-ubuntu16.04.1_amd64.deb
is not the same as the one at the address
http://security.ubuntu.com/ubuntu/pool/universe/x/xresprobe/
suggested as an alternative for wget-ing it from. There they have 
xresprobe_0.4.24ubuntu9_amd64.deb
which is a bigger 54 k file & has a last modified date of 2009-10-04 01:04.
Which seems kind of odd. That's more that 7 years ago. Which is better?
I tried both & got the same result.
`- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -`
Burning a DVD to try in the baremetal machine now. We'll see.

**Comment by ch1x0r** [comment]
Was the virtual vm a live or installed?

**Comment by Lew-Rockwell-Fan** [comment]
Thank you so  much for answering. Virtual in VirtualBox. That is why I asked if there was any point in trying to burn a DVD and try it in bare metal. You think that would be worth a try?

**Comment by ch1x0r** [comment]
I have never been able to backup or create an iso of a live whether in VM
(vbox) or bare metal.....

**Comment by Lew-Rockwell-Fan** [comment]
I think I was unclear. I did `sudo respin backup` on a baremetal system. Now I'm trying to boot a vm from the iso in order to test it before burning a DVD to boot the baremetal system.

**Comment by ch1x0r** [comment]
Ok, the iso worked? Should work on VM... Could also work in a DVD or usb... I haven't used DVD in a long time. I use USB... or vm.
Was everything ok with the backup?

**Comment by Lew-Rockwell-Fan** [comment]
Well, the iso does not boot in a VM. So I don't know if it will work in any way or not. I'm trying to make a live DVD to boot my baremetal system from (and also to have a DVD I can restore or clone with). Are you saying I should go ahead and try burning it, despite the failure in the VM?

**Comment by Lew-Rockwell-Fan** [comment]
Burned & verified the latest iso to DVD with k3b. There was 1 difference in the way bare metal booted from the DVD from what happened in the VM booting the iso:

The repeating error message was 2 lines instead of one:

Not a typewriter.
mount: mounting /dev/sr0 on /cdrom failed: No such device
Not a typewriter.
mount: mounting /dev/sr0 on /cdrom failed: No such device
. . . [the same 2 lines repeating many times]

Ending in the Busybox prompt was the same.
Should I try a thumb drive?

**Comment by ch1x0r** [comment]
Yea. Try a thumb drive. Just wondering.

**Comment by Lew-Rockwell-Fan** [comment]
I fixed the group name first. Took longer than I expected. Didn't change a thing. So much for my intuition.

From a thumb drive the error message is slightly different. The 3 line repeating pattern is:

Stdin: Not a typewriter
/init line 3: can't open /dev/sr0: No medium found
/init line 3: can't open /dev/sr0: No medium found

And after many repetitions, on to the same Busybox message.

I guess next I'll try booting in the 4.4.0-75 kernel unless you have a better idea. If that works, it isn't ideal but it would still be useful.

**Comment by Lew-Rockwell-Fan** [comment]
The most recent working live-disk backup I have yet found was made about 3 months ago with PinguyBuilder and uses the 4.4.0-63 kernel. I may still find a more recent one. I'm, . . . um . . . neatness-challenged. I'll reinstall '63 and try that. It occurs to me I also added a second WM to play with recently - 9wm. I wouldn't think that would cause this problem, but it the '63 doesn't work, I'll try to figure out to coax apt  into telling me my install/purge history and reverse recent changes a few at a time. Or maybe backup this system with fsarchiver and restore old fsa backups until I figure out in between which 2 fsa backups this broke, start from the  one before, etc., etc.

**Comment by ch1x0r** [comment]
I'm in Arizona with plans to head to Hartford CT and then Scotland. I can
go to LA, San Fran Nyc or boston...Are you anywhere near any of those
locations in the next few months to meet. If so I'd like a demo...To
understand better and possibly resolve... I might be able to get to Seattle
area also.

**Comment by Lew-Rockwell-Fan** [comment]
Well, that would be fun. But I'm in the deep South and couldn't possibly travel far right now. I have too many dependent animals. I'll keep plugging away at this.  If nothing else, restoring an fsa backup is almost certain to restore the ability to make  a respun CD. From there, I can change one thing at a time until it breaks.  Booting with the '63 kernel made no difference.

I thank you for the offer though, and hope I can meet you another time. Have good trips. Sounds like fun.

**Comment by Lew-Rockwell-Fan** [comment]
Also `sudo respin dist` has the same result.

**Comment by Lew-Rockwell-Fan** [comment]
OK, getting there. I restored an fsarchiver backup of an older (stlll 16.04 though) version of the system.

Verified that older respin works there.

Did:
apt-get update
apt-upgrade
apt-get install linux-generic

Used gdebi to install:

- xresprobe from here (not the one from Ubuntu), xresprobe_0.4.24ubuntu11-0-1-ubuntu16.04.1_amd64.deb

-The current respin, respin_2.0.6-0-73-ubuntu16.04.1_amd64.deb

Still works. OK, that eliminates a lot of possibilities.

Then I realized that going back and forth reinstalling new-sys/old-sys was going to be a a PITA, so I made room and installed the fsarchiver backup of the newer version of the system where Respin is broken to a new partition. Adjusted hostname files, sudoers, uuid, lilo.conf, fstab, etc., etc. Now I can reboot back and forth, gradually changing, step by step, the older version of the system where Respin works into a clone of the new version where it doesn't, testing Respin in between each step until I break it. Then we'll know what did it. Five possibilities that I can see:

-something pulled in with `apt-get dist-upgrade` which I haven't done yet
-something I installed
-something I purged
-some setting or config file I changed
-something that got corrupted

If it is the last, I may not be able to pin it down, but the first 4 are more likely. I'll post when I have a result.

**Comment by Lew-Rockwell-Fan** [comment]
I thought maybe it was the UNKNOWN user with gid 1002 that was causing the problem, since the output of mksquashfs continued to mention that group after I thought I'd gotten rid of it, but I think I've finally tracked down  and chowned every file that affected and eliminated all traces of group unknown:1002 and mksquashfs no longer mentions it.  But I still have  the same error booting iso files made on that system. So apparently that was a red herring.

Two things I've noticed about the error message that appears just before the splash screen in the failing boots, the one I quoted above as:

[ 2.000001] Failed to access perfecter msr (MSR c0010007 is 0)

1. The initial number varies slightly. I'm wondering if it is a time stamp relative to some event happening about 2 seconds previously.
2. It does NOT appear in booting the iso files that I've made on the other system restored from an fsarchiver backup of an earlier version of this one - in other words, the ones that work. So now I'm thinking it may be more significant than I thought, and I'll be trying to track it down.

On the restored system, I've installed almost all the packages on the broken system and copied all my scripts and all my config files from ~. And respin still works there. So as a debugging technique, so far it hasn't yielded any information yet.  On the other hand, the exercise will not be pointless in any case, as I do now have near identical clone of the system where respin was failing - a good clone where respin works. So, as a purely pragmatic matter for my own usage, the problem is solved. I could delete the good clone's evil twin. But I'd still like to know why it happened & I'm still trying to track it down.

**Comment by ch1x0r** [comment]
Definitely let us know! Actually, we really need to figure this out -

**Comment by Lew-Rockwell-Fan** [comment]
Another fact: Busy box says the group is 0/UNKNOW for the ancestral directories / and /dev/ also.  Also I read that although the Busybox that does the initramfs in 16.04 does not have chown, Busybox can be compiled with chown. I'm not sure yet how to go about replacing mine with a version that has it. Can you think of any way to determine what Busybox on a WORKING system thinks the group that owns /dev/sr0 is? I never really SEE Busybox when the system boots normally. Do you know of any way to tell Busybox "Hey, wait a minute before the last step, and before handing off the boot, let me ask you some questions."?

**Comment by Lew-Rockwell-Fan** [comment]
I haven't given up on this, though I have somewhat back-burnered it for the moment. Lots of tedious meat-space demands on my time recently. Anyway, one more datum:
I have a 3rd 16.04 system on the same drive, much more minimalist than the twins, and I ran Respin on it.  When I boot the live iso I get the 
`[ 2.000001] Failed to access perfecter msr (MSR c0010007 is 0)`
type of message just before the Plymouth (?) splash with the slow animated dots. But when I toggle that useless bit of bling off with the escape key, the boot is normal with no problems finding /dev/sr0 and it completes, giving me a normal login prompt. Startx works, etc. All that PROVES, strictly speaking, is that whatever type of error causes that message is not automatically sufficient to cause the /dev/sr0 problem also - it could still be a necessary but insufficient cause for example. But my SWAG is to estimate the chance that it is involved much lower than I did. I think I need some sort of robotic script to crawl through both systems, logging any differences in ownership, permissions, and content of files. That is probably my next step unless someone suggests or I think of, a better approach in the mean time. Either that or study busybox to figure out how to make it log stuff to real spinning rust.
